downstreamadapter/dispatcher/basic_dispatcher.go

```diff
@@ -812,6 +829,83 @@ func (d *BasicDispatcher) cancelResendTask(identifier BlockEventIdentifier) {
 
 	task.Cancel()
 	d.resendTaskMap.Delete(identifier)
+
+	if d.IsTableTriggerEventDispatcher() {
+		d.pendingACKCount.Add(-1)
+		// If there is a held DB/All block event, report it as soon as all resend tasks are ACKed.
+		// For schedule-related non-blocking DDLs, the maintainer only ACKs after scheduling is done.
+		// For schedule-related blocking DDLs, the maintainer will only begin deal with after no pending scheduling tasks.
+		// Thus, we ensure DB/All block events can genereate correct range checkers.
+		if d.pendingACKCount.Load() == 0 {
+			if holding := d.popHoldingBlockEvent(); holding != nil {
+				d.reportBlockedEventToMaintainer(holding)
+			}
+		}
```

Comment:
Is it possible that the pendingACKCount is less than zero? It's better to add a check to cover.